### PR TITLE
chore: rework default Dockerfile

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
#### What this PR does / why we need it

uses make now to unify our build process to work exactly like we expect, everytime

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-    -o /bin/ocm ./cmds/ocm/main.go
+
+ENV BUILD_FLAGS="-trimpath"
+
+RUN make bin/ocm
 
 FROM gcr.io/distroless/static-debian12:nonroot@sha256:d71f4b239be2d412017b798a0a401c44c3049a3ca454838473a4c32ed076bfea
 
-COPY --from=build /bin/ocm /usr/local/bin/ocm
+COPY --from=build /src/bin/ocm /usr/local/bin/ocm
```
Comment (frewilhelm) [inline Dockerfile:21]: Is an absolute path necessary? What about `bin/ocm`?

File: Makefile [hunk 2]
```diff
@@ -26,7 +26,10 @@ SOURCES := $(shell go list -f '{{$$I:=.Dir}}{{range .GoFiles }}{{$$I}}/{{.}} {{e
 GOPATH                                         := $(shell go env GOPATH)
 
 NOW         := $(shell date -u +%FT%T%z)
-BUILD_FLAGS := "-s -w \
+ifneq ($(BUILD_FLAGS),)
+BUILD_FLAGS := $(BUILD_FLAGS)
```
Comment (frewilhelm) [inline Makefile:30]: This looks odd^^. If `BUILD_FLAGS` is set why do we set it again?

File: Dockerfile (hunk 1 above)
Comment (jakobmoellerdev) [reply Dockerfile:21]: /src is the working directory from the other image, it is not given by default and the makefile by default puts it under /bin/ocm but we change the WORKDIR to /src

Comment (frewilhelm) [reply Dockerfile:30]: I don't understand this. You say that the makefile puts the binary in `/bin/ocm`. Thats weird. I would have expected it in `bin/ocm`. Why is changing the `WORKDIR` to `/src` causing the binary to be in `/src/bin/ocm` Does the makefile consider the working-directory?

**Pull request closed without merging**